tests/e2e/statefulsets.go

```diff
@@ -639,6 +655,163 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-supervisor] [csi-block-vanilla
 
 	})
 
+	/*
+	  Verify List volume Response on vsphere-ccsi-controller logs
+	  Note: ist volume Threshold is set to 1 , and query limit set to 3
+	  1. Create SC
+	  2. Create statefull set with 3 replica
+	  3. Bring down the CSI driver replica to 1 , so that it is easy to validate the List volume Response.
+	  4. Wait for all the PVC to reach bound and PODs to reach running state.
+	  5. Note dow the PV volume handle
+	  6. Verify the Listvolume response in logs. It should contain all the 3 volumeID's noted in step 5
+	  7. Scale up the Statefullset replica to 5 and validate the Pagination.
+	    The 1st List volume Response will have the "token for next set:"
+	  8. Delete 2 volumes from the CNS , verify the error message in the controller logs
+	  9. Delete All the volumes
+	  10. Verify list volume response for 0 volume.
+	  11. Clean up the statefull set
+	  12. Inncrease the CSI driver  replica to 3
+
+	*/
+	ginkgo.It("[csi-block-vanilla] ListVolumeResponse Validation", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var volumesBeforeScaleUp []string
+		containerName := "vsphere-csi-controller"
+		ginkgo.By("Creating StorageClass for Statefulset")
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+			scParameters = nil
+			storageClassName = "nginx-sc-default"
+		} else {
+			ginkgo.By("CNS_TEST: Running for WCP setup")
+			profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+			scParameters[scParamStoragePolicyID] = profileID
+			storageClassName = defaultNginxStorageClassName
+			// create resource quota
+			createResourceQuota(client, namespace, rqLimit, defaultNginxStorageClassName)
+		}
```

Comment:
Looks like test is written only for block-vanilla - does this code required then ?

Reply:
List volumes applies for WCP also , but for now FSS is disabled due to a defect. Hence i didn't add csi-supervisor tag.